### PR TITLE
forms

### PR DESCRIPTION
**Pull request by Arzamas-ka**
2 - Installed formik; 
3 - Implementation of “Add movie” form with validation; 
4 - Implementation of “Edit movie” form with validation; 
5 - Use hooks from formik. 

[Киношка :)](https://drive.google.com/file/d/1zdL_hGLsozJb4P_WKSNLFlR9c6L4u5fx/view)

### CODE REVIEW

File: src/components/Select/index.tsx [hunk 1]
```diff
@@ -1,27 +1,59 @@
...
 import { SELECT_OPTIONS } from '@constants';
 
-const Select: FC<SelectProps> = ({ onChange, name }) => {
-  const options = SELECT_OPTIONS.map((option) => (
-    <option key={shortid.generate()}>{option}</option>
-  ));
+const Select: FC<SelectProps> = ({ selected, onChange }) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOnSelected = (selectedOption) => {
```
Comment (osonovaakylai) [inline src/components/Select/index.tsx:18]: useCallback is needed

File: src/reducers/movies.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        items: payload.data,
+      };
+
+    case ERROR_MOVIES:
+      return {
+        ...state,
+        error: true,
+      };
+
+    case LOADING_MOVIES:
```
Comment (osonovaakylai) [inline src/reducers/movies.ts:95]: Do you actually need this code? In every request, it is recomended to return loading and error values

File: src/actions/types.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+export const ERROR_MOVIES = 'ERROR_MOVIES';
+export const LOADING_MOVIES = 'LOADING_MOVIES';
+export const FETCH_LOAD_MORE_MOVIES = 'FETCH_LOAD_MORE_MOVIES';
+export const DELETE_MOVIE = 'DELETE_MOVIE';
+export const POSTER_ID = 'POSTER_ID';
+export const ADD_MOVIE_UI = 'ADD_MOVIE_UI';
+export const EDIT_MOVIE = 'EDIT_MOVIE';
+export const EDIT_MOVIE_UI = 'EDIT_MOVIE_UI';
+export const FILTER_MOVIES = 'FILTER_MOVIES';
+export const SORT_BY_MOVIES = 'SORT_BY_MOVIES';
```
Comment (osonovaakylai) [inline src/actions/types.ts:11]: I would refactor this part a bit. Actually we need less actions. Let's discuss it in the QA session

File: src/api/index.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+export const getSortByMovies = (sortItem) => (dispatch) => {
+  axios
+    .get(`${API_SORT_BY}${sortItem}`)
+    .then(({ data }) => {
+      dispatch(filterMovies(data));
+    })
+    .catch((error) => {
+      dispatch(errorMovies());
+    });
```
Comment (osonovaakylai) [inline src/api/index.ts:102]: How about creating reusable custom hook for fetching data. For example: useApiCall

File: src/reducers/movies.ts (hunk 2 above)
Comment (Arzamas-ka) [reply src/reducers/movies.ts:95]: Yes, this code needs to show <Spinner />
e.g.
image

File: src/components/Select/index.tsx (hunk 1 above)
Comment (Arzamas-ka) [reply src/components/Select/index.tsx]: done

File: src/reducers/movies.ts (hunk 2 above)
Comment (Arzamas-ka) [reply src/reducers/movies.ts]: removed

File: src/actions/types.ts (hunk 3 above)
Comment (Arzamas-ka) [reply src/actions/types.ts]: refactored

File: src/api/index.ts (hunk 4 above)
Comment (Arzamas-ka) [reply src/api/index.ts]: created `useApiRequest`

File: src/reducers/movies.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  switch (type) {
+    case FETCH_MOVIES:
+      return {
+        ...state,
+        items: [...state.items, ...payload.data],
+        currentPage: payload.offset + 1,
+        totalPages: Math.floor(
+          payload.totalAmount / payload.limit - payload.offset,
+        ),
+        error: null,
```
Comment (osonovaakylai) [inline src/reducers/movies.ts:31]: Does the error property is useful if error handled only with alert in useApiCall?

**Review by osonovaakylai — APPROVED** [review]

File: src/components/CongratulationsPopup/style.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+export const StyledCongratulationsPopupTitle = styled.h2`
+  margin: 0 auto;
+  padding-top: 40px;
+
+  font-size: 30px;
+  font-weight: 400;
+  color: ${colors.white};
+  text-transform: uppercase;
+  letter-spacing: 1ю2;
```
[the remark this answers is not in the document]
Comment (Arzamas-ka) [reply src/components/CongratulationsPopup/style.ts]: fixed

File: src/reducers/movies.ts (hunk 5 above)
Comment (Arzamas-ka) [reply src/reducers/movies.ts]: iIsn't useful
done

**Pull request merged**